app/src/main/java/se/sigmaconnectivity/blescanner/service/BleScanService.kt

```diff
@@ -115,53 +122,47 @@ class BleScanService() : Service() {
                 ByteBuffer.allocate(8).array()
             )
             .build()
-        compositeDisposable.add(
-            rxBleClient.scanBleDevices(scanSettings, scanFilter)
-                .doOnSubscribe {
-                    Timber.d("scanLeDevice started")
-                    scanStatus = FeatureStatus.ACTIVE
+        rxBleClient.scanBleDevices(scanSettings, scanFilter)
+            .doOnSubscribe {
+                Timber.d("scanLeDevice started")
+                scanStatus = FeatureStatus.ACTIVE
+            }
+            .doOnDispose { scanStatus = FeatureStatus.INACTIVE }
+            .subscribe(
+                { scanResult ->
+                    val timestampMillis = Duration.ofNanos(scanResult.timestampNanos).toMillis()
+                    assembleUID(scanResult)?.let {
+                        if (scanResult.callbackType == ScanCallbackType.CALLBACK_TYPE_FIRST_MATCH) {
+                            processFirstMatch(it, timestampMillis)
+                        } else {
+                            processMatchLost(it, timestampMillis)
+                        }
+                    } ?: Timber.e("Can not assemble UID")
+                },
+                {
+                    Timber.d("Device found with error \n $it")
                 }
-                .doOnDispose { scanStatus = FeatureStatus.INACTIVE }
-                .subscribe(
-                    {scanResult ->
-                        val timestampMillis = Duration.ofNanos(scanResult.timestampNanos).toMillis()
-                        assembleUID(scanResult)?.let {
-                            if (scanResult.callbackType == ScanCallbackType.CALLBACK_TYPE_FIRST_MATCH) {
-                                processFirstMatch(it, timestampMillis)
-                            } else {
-                                processMatchLost(it, timestampMillis)
-                            }
-                        } ?: Timber.e("Can not assemble UID")
-                    },
-                    {
-                        Timber.d("Device found with error \n $it")
-                    }
-                )
-        )
+            ).addTo(compositeDisposable)
     }
 
     private fun processFirstMatch(contactHash: String, timestamp: Long) {
         Timber.d("CALLBACK_TYPE_FIRST_MATCH: $contactHash")
-        compositeDisposable.add(
-            contactUseCase.processContactMatch(contactHash, timestamp)
-                .subscribe({
-                    Timber.d("processContactMatch() SUCCESS")
-                }, {
-                    Timber.e("processContactMatch() FAILED \n $it")
-                })
-        )
+        contactUseCase.processContactMatch(contactHash, timestamp)
+            .subscribe({
+                Timber.d("processContactMatch() SUCCESS")
+            }, {
+                Timber.e("processContactMatch() FAILED \n $it")
```

Comment:
```suggestion
                Timber.e(it, "processContactMatch() FAILED")
```
To tylko mała sugestia, ale tak wydrukuje ładniej ze stacktrace